recipes/ncurses/all/conanfile.py

```diff
@@ -0,0 +1,180 @@
+        conf_args.extend([
+            "--enable-reentrant" if self.options.with_reentrant else "--disable-reentrant",
+            "--enable-widec" if self.options.with_widec else "--disable-widec",
+            "--with-pcre2" if self.options.with_pcre2 else "--without-pcre2",
+            "--without-libtool",
+            "--without-ada",
+            "--without-manpages",
+            "--without-tests",
+            "--disable-echo",
+            "--without-debug",
+            "--without-profile",
+            "--with-sp-funcs",
+            "--disable-rpath",
+            "--datarootdir={}".format(tools.unix_path(os.path.join(self.package_folder, "bin", "share"))),
+            "--disable-pc-files",
+        ])
+        if self.settings.os == "Windows":
+            conf_args.extend([
+                "--disable-macros",
+                "--disable-termcap",
+                "--enable-database",
+                "--enable-sp-funcs",
+                "--enable-term-driver",
+                "--enable-interop",
+            ])
+        if self.settings.compiler == "Visual Studio":
+            conf_args.extend([
+                "ac_cv_func_getopt=yes",
+            ])
+            build = host = "{}-w64-mingw32-msvc7".format(self.settings.arch)
+            self._autotools.flags.append("-FS")
+            self._autotools.cxx_flags.append("-EHsc")
+        self._autotools.configure(args=conf_args, configure_dir=self._source_subfolder, host=host, build=build)
+        return self._autotools
+
+    def _patch_sources(self):
+        for patch in self.conan_data["patches"][self.version]:
+            tools.patch(**patch)
+
+    @contextmanager
+    def _build_context(self):
+        if self.settings.compiler == "Visual Studio":
+            with tools.vcvars(self.settings):
+                msvc_env = {
+                    "CC": "cl -nologo",
+                    "CXX": "cl -nologo",
+                    "LD": "link -nologo",
+                    "LDFLAGS": "user32.lib",
+                    "NM": "dumpbin -symbols",
+                    "STRIP": ":",
+                    "AR": "lib -nologo",
+                    "RANLIB": ":",
+                }
+                with tools.environment_append(msvc_env):
+                    yield
+        else:
+            yield
+
+    def build(self):
+        self._patch_sources()
+        with self._build_context():
+            autotools = self._configure_autotools()
+            autotools.make(target="libs" if self.settings.os == "Windows" else None)
+
+    @property
+    def _major_version(self):
+        return self.version.split(".")[0]
+
+    def package(self):
+        self.copy("COPYING", src=self._source_subfolder, dst="licenses")
+        with self._build_context():
+            autotools = AutoToolsBuildEnvironment(self, win_bash=tools.os_info.is_windows)
+            autotools.make(target="install.libs" if self.settings.os == "Windows" else "install")
```

Comment:
Reminder to self: revisit install targets

Reply:
This is needed.
`progs` do currerntly not build for Visual Studio.
I think this is ok as this recipe concentrates on the libraries.
I'll follow up my msvc patches upstream and see what is needed to make these work.
For the moment, this is ok to merge in CCI.